doc/api/worker_threads.md

```diff
@@ -1329,6 +1329,228 @@ from the running process and will preload the same preload scripts as the main
+return directly from a `.runLoop()`, `.runLoopUntilPromiseResolved()` or
+`.runInWorkerScope()` call. This will render the Node.js instance unusable
+and is generally comparable to running `process.exit()`.
+
+This method returns a `Promise` that will be resolved when all resources
+associated with this Node.js instance are released. This `Promise` resolves on
+the event loop of the _outer_ Node.js instance.
+
+#### `synchronousWorker.createRequire(filename)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `filename` {string}
+
+Create a `require()` function that can be used for loading code inside the
+inner Node.js instance.
+
+#### `synchronousWorker.globalThis`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {Object}
+
+Returns a reference to the global object of the inner Node.js instance.
+
+#### `synchronousWorker.process`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {Object}
+
+Returns a reference to the `process` object of the inner Node.js instance.
+
+### FAQ
+
+#### What does a SynchronousWorker do?
+
+Creates a new Node.js instance, using the same thread and the same JS heap.
+You can create Node.js API objects, like network sockets, inside the new
+Node.js instance, and spin the underlying event loop manually.
+
+#### Where did SynchronousWorker come from?
+
+`SynchronousWorker` was originally developer by Node.js core contributor
+Anna Henningsen and published as a separate module [`synchronous-worker`][] on
+npm under the MIT license. It was integrated into Node.js core with Anna's
+permission. The majority of the code, documentation, and tests were adopted
+almost verbatim from the original module.
+
+#### Why would I use a SynchronousWorker?
+
+The most common use case is probably running asynchronous code synchronously,
+in situations where doing so cannot be avoided (even though one should try
+really hard to avoid it). Another popular npm package that does this is
+[`deasync`][], but `deasync`
+
+* solves this problem by starting the event loop while it is already running
+  (which is explicitly _not_ supported by libuv and may lead to crashes)
+* doesn’t allow specifying _which_ resources or callbacks should be waited for,
+  and instead allows everything inside the current thread to progress.
+
+#### How can I avoid using SynchronousWorker?
+
+If you do not need to directly interact with the objects inside the inner
+Node.js instance, a lot of the time Worker threads together with
+[`Atomics.wait()`][] will give you what you need.
+
+#### My async functions/Promises/… don’t work
+
+If you run a `SynchronousWorker` with its own microtask queue (i.e. in default
+mode), code like this will not work as expected:
+
+```cjs
+const { SynchronousWorker } = require('node:worker_threads');
+const w = new SynchronousWorker();
+let promise;
+w.runInWorkerScope(() => {
+  promise = (async () => {
+    return w.createRequire(__filename)('node-fetch')('...');
+  })();
+});
+w.runLoopUntilPromiseResolved(promise);
+```
```

Comment:
Adding the ESM version. Also could we rewrite this example to use the `fetch` that’s now part of core, rather than `node-fetch`?

```suggestion
```mjs
import { SynchronousWorker } from 'node:worker_threads';
const w = new SynchronousWorker();
let promise;
w.runInWorkerScope(() => {
  promise = (async () => {
    return w.createRequire(import.meta.url)('node-fetch')('...');
  })();
});
w.runLoopUntilPromiseResolved(promise);
```
```cjs
const { SynchronousWorker } = require('node:worker_threads');
const w = new SynchronousWorker();
let promise;
w.runInWorkerScope(() => {
  promise = (async () => {
    return w.createRequire(__filename)('node-fetch')('...');
  })();
});
w.runLoopUntilPromiseResolved(promise);
```
```

Reply:
As I note in the test, there is something stopping this from working with the built-in fetch that I do not quite yet understand. Before we update the documentation to use the internal fetch we'll need to investigate that issue (which should not need to be done before this lands)

Reply:
I specifically left the mjs examples out for now because I want to do further testing to validate those cases before adding the examples.